src/ray/raylet/scheduling/policy/node_scheduling_policy.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/raylet/scheduling/policy/node_scheduling_policy.h"
+
+namespace ray {
+namespace raylet_scheduling_policy {
+
+scheduling::NodeID NodeSchedulingPolicy::Schedule(const ResourceRequest &resource_request,
+                                                  SchedulingOptions options) {
+  RAY_CHECK(options.scheduling_type == SchedulingType::NODE);
+
+  for (const auto &[node_id, node] : nodes_) {
+    if (node_id.Binary() != options.node_id) {
+      continue;
+    }
+    if (!is_node_available_(node_id)) {
```

Comment:
Does this mean we'd only schedule to the node if it has resources available?

Reply:
I think we'd want to queue it on that remote node in that case. So maybe this logic can be simplified to: `node_exists ? node_id : soft ? hybrid() : Nil`?

Reply:
`is_node_available_` is about whether this node is alive or not. It has nothing to do with resources. (the variable name can be improved though)

Reply:
ah, should name it `is_node_alive_``

Reply:
Agree. I'll make a follow-up PR to fix it, don't want to make the current one too big.

Reply:
Looks like we could rename it for just this class in this PR.